lib/QrReader.php

```diff
@@ -62,7 +62,7 @@ public function __construct($imgSource, $sourceType = QrReader::SOURCE_TYPE_FILE
             $height = $im->getImageHeight();
             $source = new IMagickLuminanceSource($im, $width, $height);
         } else {
-            if (!is_resource($im)) {
+            if (!is_resource($im) && !is_object($im)) {
```

Comment:
Resource to object change in PHP 8

Reply:
I think it should be done like this actually, for proper type handling

```suggestion
            if ($im === false) {
```

PHPStan would complain: https://phpstan.org/r/15d5bc46-209a-4f2c-bc01-5e29daddb8fc

With `=== false`: https://phpstan.org/r/e1632f88-9ea3-45ef-84e6-86cbbaddc649

Reply:
Sorry, but I really think phpstan this warning have no value here.

Reply:
I'm not stating PHPStan as the reason, but rather as an example. This is a simpler check that actually targets the correct type returned in both PHP 7 and 8. I don't see a reason not to simplify it here.